site/en/_posts/2020-07-15-ocrd-workspace-cli.md

```diff
@@ -0,0 +1,207 @@
+```
+
+* Create a workspace at `/tmp/ws` from a remote METS URL:
+
+```sh
+ocrd workspace --directory /tmp/ws clone https...
+```
+
+* Create a workspace at `/tmp/ws` from a remote URL, **also downloading all files referenced in the METS**:
+
+```sh
+ocrd workspace --directory /tmp/ws clone --download https...
+```
+
+## Add
+
+Once you have a workspace, i.e. a `mets.xml` file in place, you can use the
+command line tools to do all kinds of operations on the workspace. Here we'll
+focus on how to add files to the workspace.
+
+Files can be added with the `ocrd workspace add` command. When adding a file,
+there are a number of required parameters to describe the file, as per its help:
+```sh
+ocrd workspace add --help
+Usage: ocrd workspace add [OPTIONS] FNAME
+
+  Add a file or http(s) URL FNAME to METS in a workspace. If FNAME is not an
+  http(s) URL and is not a workspace-local existing file, try to copy to
+  workspace.
+
+Options:
+  -G, --file-grp TEXT      fileGrp USE  [required]
+  -i, --file-id TEXT       ID for the file  [required]
+  -m, --mimetype TEXT      Media type of the file  [required]
+  -g, --page-id TEXT       ID of the physical page
+  -C, --check-file-exists  Whether to ensure FNAME exists
+  --ignore                 Do not check whether file exists.
+  --force                  If file with ID already exists, replace it. No
+                           effect if --ignore is set.
+
+  --help                   Show this message and exit.
+```
+
+The `-G`, `-i` and `-m` options are required and we strongly recommend you also
+always set the `--page-id` for a file to make it clear which page the file
+being added represents.
+
+For example, let's assume you have a PNG image at `/tmp/IMG_0001`, representing
+the first physical page of the work. You can add that to the empty workspace at `/tmp/ws` [we created before](#init-and-clone), like so:
+
+```sh
+ocrd workspace --directory /tmp/ws add -i FILE_0001_PNG -G OCR-D-IMG -m image/png -g PHYS_0001 /tmp/IMG_0001.png
+```
+
+Here's a breakdown of the options
+
+* `-i FILE_0001_PNG`: This must be a unique identifier of a file. Make sure to make it specific enough, e.g. by adding the format (as we're doing here) or with some other consistent naming scheme
+* `-G OCR-D-IMG`: Add this file to the `OCR-D-IMG` file group (or create it if it does not yet exist). The names of `mets:fileGrp` are **completely arbitrary**. By convention, OCR-D uses `OCR-D-IMG` for the unprocessed images in [the Ground Truth](https://ocr-d.de/gt-repo/) but other conventions call this file group `MAX` (for MAXimum resolution).
+* `-m image/png`: The media type or MIME type of the file, `image/png` for PNG images, `image/tiff` for TIFF, etc.
+* `-g PHYS_0001`: The ID of the physical page, i.e. the `mets:div` that groups together all files representing the same page. The identifier is arbitrary as well, though many METS tools chose `PHYS_` + numerical identifier as the naming convention.
+
```

Comment:
```suggestion
Note that -i and -g always have to start with a letter
```

Reply:
ah well spotted, I wanted to add that but forgot.